server/jwt/jwt.go

```diff
@@ -0,0 +1,115 @@
+package jwt
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/fabiocicerchia/go-proxy-cache/config"
+	"github.com/fabiocicerchia/go-proxy-cache/server/handler"
+	"github.com/fabiocicerchia/go-proxy-cache/utils/slice"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+var co *config.Jwt
+
+func errorJson(resp http.ResponseWriter, statuscode int, error *config.JwtError) {
+	resp.WriteHeader(statuscode)
+	resp.Header().Add("Content-Type", "application/json; charset=utf-8")
+	json_error, _ := json.Marshal(error)
+	resp.Write(json_error)
+}
+
+func logJWTErrorAndAbort(w http.ResponseWriter, err error) error {
+	co.Logger.Info("Error jwt:", err)
+	errorJson(w, http.StatusUnauthorized, &config.JwtError{ErrorCode: "JsonWebTokenError", ErrorDescription: err.Error()})
+
+	return http.ErrAbortHandler
+}
+
+func ValidateJWT(w http.ResponseWriter, r *http.Request, keySet jwk.Set) error {
+	token, err := jwt.ParseRequest(r,
+		jwt.WithKeySet(keySet),
+		jwt.WithValidate(true),
+		jwt.WithTypedClaim("scope", json.RawMessage{}),
+		jwt.WithTypedClaim("scp", json.RawMessage{}),
+	)
+	if err != nil {
+		return logJWTErrorAndAbort(w, err)
+	}
+	if err := jwt.Validate(token); err != nil {
+		return logJWTErrorAndAbort(w, err)
```

Comment:
it would be good to increase the test coverage by adding a test case to cover this scenario as well (ie testing the failure of validation).